### PR TITLE
feat(pt_expt): atomic model 

### PR DESCRIPTION
**Pull request by wanghan-iapcm**
## Summary by CodeRabbit

* **New Features**
  * Added end-to-end bias/statistics workflow for atomic models (compute, load, apply, and update output biases).
  * Introduced PyTorch-experimental atomic model wrappers with serialization/export compatibility.
  * Added comprehensive statistics utilities for global and per-atom outputs.

* **Bug Fixes**
  * Improved tensor→array conversion to handle gradient-enabled tensors robustly.

* **Tests**
  * Added extensive tests covering stats, bias workflows, serialization, export, and consistency.

### CODE REVIEW

File: deepmd/dpmodel/atomic_model/base_atomic_model.py [hunk 1]
```diff
@@ -246,6 +252,196 @@ def call(
...
+            fparam: np.ndarray | None = None,
+            aparam: np.ndarray | None = None,
+        ) -> dict[str, np.ndarray]:
+            # Get reference array to determine the target array type and device
+            # Use out_bias as reference since it's always present
+            ref_array = self.out_bias
+            xp = array_api_compat.array_namespace(ref_array)
+
+            # Convert numpy inputs to the model's array type with correct device
+            device = getattr(ref_array, "device", None)
```
Comment (njzjz) [inline deepmd/dpmodel/atomic_model/base_atomic_model.py]:
```suggestion
            device = array_api_compat.device(ref_array)
```

File: deepmd/dpmodel/atomic_model/base_atomic_model.py [hunk 2]
```diff
@@ -246,6 +252,196 @@ def call(
...
+                atype = xp.asarray(atype)
+                if box is not None:
+                    if np.allclose(box, 0.0):
+                        box = None
+                    else:
+                        box = xp.asarray(box)
+                if fparam is not None:
+                    fparam = xp.asarray(fparam)
+                if aparam is not None:
+                    aparam = xp.asarray(aparam)
```
Comment (njzjz) [inline deepmd/dpmodel/atomic_model/base_atomic_model.py]: This can be removed since `device` will never be `None`.

File: deepmd/pt_expt/atomic_model/dp_atomic_model.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+class DPAtomicModel(DPAtomicModelDP, torch.nn.Module):
+    # Import at class level to set base classes for deserialization
+    # These will be used by the dpmodel deserialize method to create pt_expt instances
+    from deepmd.pt_expt.descriptor.base_descriptor import (
+        BaseDescriptor,
+    )
+    from deepmd.pt_expt.fitting.base_fitting import (
+        BaseFitting,
+    )
```
Comment (njzjz) [inline deepmd/pt_expt/atomic_model/dp_atomic_model.py]: Why does it need to import in the class?

**Review by njzjz — APPROVED** [review]

**Review by njzjz — APPROVED** [review]

**Review by iProzd — APPROVED** [review]
